Controller/NotifyHandler.php

```diff
@@ -74,10 +74,13 @@ public function handleRequest()
         }
 
         // Return the response or log errors if any happen.
-        if ($oNotificationResponse instanceof SuccessResponse && $oNotificationResponse->isValidSignature()) {
-            $this->_onNotificationSuccess($oNotificationResponse, $oService);
-        } else {
-            $this->_onNotificationError($oNotificationResponse);
+        $oTransaction = oxNew(Transaction::class);
+        if (!$oTransaction->loadWithTransactionId($oNotificationResponse->getTransactionId())) {
```

Comment:
Maybe we could do this in a slightly more readable way, for instance like this?

```suggestion
        if ($oTransaction->loadWithTransactionId($oNotificationResponse->getTransactionId())) {
        	// if a transaction with this ID already exists we do not need to handle it again
        	return;
        }
        
        if ($oNotificationResponse instanceof SuccessResponse && $oNotificationResponse->isValidSignature()) {
```

Reply:
👍